client/client.go

```diff
@@ -365,6 +374,17 @@ func (c *Client) RegisterProtobufs(ctx context.Context, in *service.RegisterProt
 	return errors.WithStack(err)
 }
 
+func (c *Client) getTLSDialOptions() (grpc.DialOption, error) {
+	if len(c.tlsConfig.CACert.Contents) == 0 {
+		return grpc.WithInsecure(), nil
```

Comment:
Defaulting to no TLS if no CA cert?

Reply:
Fixed this. Now it checks the API TLS config is not empty to decide to turn TLS on/off.